source/kraken/fill_disruption_from_chaos.cpp

```diff
@@ -291,6 +313,90 @@ struct add_impacts_visitor : public apply_impacts_visitor {
         }
         return true;
     }
+
+    void operator()(const nt::StopArea* stop_area) override {
+        for (const auto stop_point : stop_area->stop_point_list) {
+            // We block all the journey pattern of the stop_area according to the impact
+            for (const auto jpp : stop_point->journey_pattern_point_list) {
+                apply_impacts_visitor::operator()(jpp->journey_pattern);
+            }
+            if (stop_point->journey_pattern_point_list.empty() ||
+                    stop_point->journey_pattern_point_list.front()->journey_pattern == nullptr) {
```

Comment:
The first one yep.
The second one could happen if you disrupt all the stop points of a route.